xcube/core/store/descriptor.py

```diff
@@ -141,36 +169,55 @@ def __init__(self,
                          time_range=time_range,
                          time_period=time_period,
                          open_params_schema=open_params_schema)
+        self.coords = coords
         self.dims = dict(dims) if dims else None
-        self.data_vars = list(data_vars) if data_vars else None
+        if data_vars:
+            if type(data_vars) == dict:
+                self.data_vars = data_vars
+            else:
+                self.data_vars = {}
+                for data_var in data_vars:
+                    self.data_vars[data_var.name] = data_var
+        else:
+            self.data_vars = None
         self.attrs = _convert_nans_to_none(dict(attrs)) if attrs else None
 
     @classmethod
     def from_dict(cls, d: Mapping[str, Any]) -> 'DatasetDescriptor':
         """Create new instance from a JSON-serializable dictionary"""
         assert_in('data_id', d)
-        data_vars = []
-        data_vars_as_dicts = d.get('data_vars', [])
-        for data_var_as_dict in data_vars_as_dicts:
-            data_vars.append(VariableDescriptor.from_dict(data_var_as_dict))
+        data_vars_dict = None
+        data_vars = d.get('data_vars', None)
+        if data_vars:
+            data_vars_dict = {}
+            if type(data_vars) == dict:
```

Comment:
Insufficient condition. 
```suggestion
            if instanceof(data_vars, Mapping):
```